src/simtools/ray_tracing/psf_parameter_optimisation.py

```diff
@@ -0,0 +1,794 @@
+        elif param_name in ["mirror_align_random_horizontal", "mirror_align_random_vertical"]:
+            # [deg, deg, dimensionless, dimensionless]
+            psf_pars_with_units[param_name] = [
+                param_values[0] * u.deg,
+                param_values[1] * u.deg,
+                param_values[2] * u.dimensionless_unscaled,
+                param_values[3] * u.dimensionless_unscaled,
+            ]
+        else:
+            # For any other parameters, keep as-is
+            psf_pars_with_units[param_name] = param_values
+
+    return psf_pars_with_units
+
+
+def export_psf_parameters_as_json(best_pars, tel_model, parameter_version, output_dir, func_logger):
+    """
+    Export PSF parameters as JSON model parameter files.
+
+    Parameters
+    ----------
+    best_pars : dict
+        Best parameter set
+    tel_model : TelescopeModel
+        Telescope model object
+    parameter_version : str
+        Parameter version string
+    output_dir : Path
+        Output directory path
+    func_logger : Logger
+        Logger object
+    """
+    try:
+        func_logger.info("Exporting best PSF parameters as JSON model parameter files")
+        psf_pars_with_units = _add_units_to_psf_parameters(best_pars)
+        parameter_output_path = output_dir / tel_model.name
+        for parameter_name, parameter_value in psf_pars_with_units.items():
+            writer.ModelDataWriter.dump_model_parameter(
+                parameter_name=parameter_name,
+                value=parameter_value,
+                instrument=tel_model.name,
+                parameter_version=parameter_version,
+                output_file=f"{parameter_name}-{parameter_version}.json",
+                output_path=parameter_output_path,
+                use_plain_output_path=True,
+            )
+        func_logger.info(f"JSON model parameter files exported to {output_dir}")
+    except ImportError as e:
+        func_logger.warning(f"Could not export JSON parameters: {e}")
+    except (ValueError, KeyError, OSError) as e:
+        func_logger.error(f"Error exporting JSON parameters: {e}")
+
+
+def run_psf_optimization_workflow(tel_model, site_model, args_dict, output_dir, func_logger):
+    """
+    Run the complete PSF parameter optimization workflow.
+
+    This function consolidates the main optimization logic to make the application lighter.
+
+    Parameters
+    ----------
+    tel_model : TelescopeModel
+        Telescope model object
+    site_model : SiteModel
+        Site model object
+    args_dict : dict
+        Dictionary containing parsed command-line arguments
+    output_dir : Path
+        Output directory path
+    func_logger : Logger
+        Logger object
+
+    Returns
+    -------
+    None
+        All results are saved to files and printed to console
+    """
+    # Generate parameter sets
+    all_parameters = []
+    mrra_0, mfr_0, mrra2_0, mar_0 = get_previous_values(tel_model)
+
+    n_runs = 5 if args_dict["test"] else 50
```

Comment:
Do you have an idea where the 50 come from? Does this need to be configurable?

Reply:
Let's talk offline and you can remind me how the optimisation works. Not sure if this is the max number of runs?